### PR TITLE
Return task from loadJSON, postJSON

### PR DESCRIPTION
**Pull request by 1ec5**
This gives client code the opportunity to cancel the request if necessary.

Along the way, I also introduced a couple `guard` statements to make the code flow a little easier to follow.

### CODE REVIEW

**Comment by 1ec5** [comment]
Not sure what the Travis error is about, but it seems to be unrelated to this change.

**Comment by pietbrauer** [comment]
It's because some gem was yanked. My PR #13 updated it already. This needs some rework with the new protocol based implementation anyways.

**Comment by 1ec5** [comment]
Ooh, I hadn’t noticed that PR – that’s great. I’ll rebase once it lands.

**Comment by pietbrauer** [comment]
@1ec5 merged #13

**Comment by 1ec5** [comment]
Sorry for the delay, @pietbrauer. I’ve rebased the change onto 708b0bce9b4cd1dabf1f74fd9a69519d0fb17b62 on master. Let me know if I’ve overlooked anything.

File: RequestKit/JSONPostRouter.swift [hunk 1]
```diff
@@ -1,40 +1,46 @@
...
+            data = try NSJSONSerialization.dataWithJSONObject(params, options: NSJSONWritingOptions())
+        } catch {
+            completion(json: nil, error: error)
+            return nil
+        }
 
-                    if let error = error {
+        let task = session.uploadTaskWithRequest(request, fromData: data) { data, response, error in
+            if let response = response as? NSHTTPURLResponse {
+                if response.statusCode != 201 {
```
Comment (pietbrauer) [inline RequestKit/JSONPostRouter.swift:23]: Can you use `response.wasSuccessful == false` instead of checking the code?

**Comment by pietbrauer** [comment]
Sorry this took so long, I was on vacation. Just one 💬 and it looks good. Thanks a lot!

File: RequestKit/JSONPostRouter.swift (hunk 1 above)
Comment (pietbrauer) [inline RequestKit/JSONPostRouter.swift:23]: Handled in https://github.com/nerdishbynature/RequestKit/commit/bf2cc97b98f876e1564877186a0171054787f27c

Comment (1ec5) [inline RequestKit/JSONPostRouter.swift:23]: Ah, I missed your comment. Thanks for taking care of it.

**Pull request merged**